.github/workflows/cmake.yml

```diff
@@ -41,6 +41,13 @@ jobs:
             cmake-args: -DWITH_NATIVE_INSTRUCTIONS=ON
             codecov: ubuntu_gcc_native_inst
 
+          - name: Ubuntu GCC Native Instructions (AVX)
+            os: ubuntu-latest
+            compiler: gcc
+            cxx-compiler: g++
+            cmake-args: -DWITH_NATIVE_INSTRUCTIONS=ON -DNATIVEFLAG=-mavx
```

Comment:
I am not sure it is a good idea to export `NATIVEFLAG` directly like this.

I think exporting for example `NATIVE_ARCH_OVERRIDE` instead, and setting `NATIVEFLAG` based on it would be better. And perhaps print a warning in/after the CMake summary (Like CPack does on line :-4 at the end of CMakeLists.txt).
`message (WARNING "WITH_NATIVE_INSTRUCTIONS enabled, but running with NATIVE_ARCH_OVERRIDE: XX")`

The option is great for testing, and potentially good for making very narrow builds for a different machine, but also very much an advanced option that can also break things in subtle or not-so-subtle ways.
With the renamed parameter and the warning, I think it makes it more clear to potential users that this is not for the casual tinkerer and also easier to understand what its purpose is.

What do you think @phprus?

Reply:
@Dead2 
``NATIVE_ARCH_OVERRIDE`` is a good name.

I think it's better to print ``message(STATUS ...)`` instead of ``message(WARNING ...)``.

PR updated.